### ISSUE TITLE
BUG: save(query("test.jld"),a) is an endless loop 

### ISSUE THREAD

**Issue by zhmz90** [JuliaLang/JLD.jl#46]
```
function save{F}(q::Formatted{F}, data...; options...)
    unknown(q) && throw(UnknownFormat(q))
    libraries   = applicable_savers(q)
    failures    = Any[]
    for library in libraries
        try
            Library = checked_import(library)
            return Library.save(q, data...; options...)
        catch e
            handle_current_error(e, library, library == last(libraries))
            push!(failures, (e, q))
        end
    end
    handle_error(failures)
end
```

When you execute the below code, an endless loop comes because `Library` is `JLD`.

```
using JLD
using FileIO
a = 1
save(query("test.jld"), a)
```

```
julia> versioninfo()
Julia Version 0.5.0-dev+1516
Commit a7b41f5 (2015-11-30 00:08 UTC)
Platform Info:
  System: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Xeon(R) CPU E5-2699 v3 @ 2.30GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.3
```

**Comment by zhmz90** [comment]
Is there anyone else who encountered the same problem?

**Comment by timholy** [comment]
You need to provide an HDF5-name for `a`, for example
```jl
save(query("test.jld"), "a", a)
```

**Comment by zhmz90** [comment]
@timholy Thank you, very much.

**Comment by timholy** [comment]
No problem.

Out of curiosity, is there any reason you're calling `query` directly? You should just be able to say `save("test.jld", "a", a)`.

**Comment by zhmz90** [comment]
I just remembered the wrong style  ```save("test.jld",a)``` .  When I try to debug it, called ```query```.

**Comment by timholy** [comment]
Thanks for explaining.

It would be good to have this give an error rather than segfault/hang. I opened https://github.com/JuliaLang/julia/issues/14472, which appears necessary to fix the problem in a backward-compatible way.

**Comment by zhmz90** [comment]
That's great! Looking forward to.

**Comment by mweastwood** [comment]
Can this be reopened until it is fixed (ie. turned into an error)? I think this is a common typo and I was just about to open a duplicate issue.

**Comment by timholy** [comment]
Closed by #50
